pkg/server/router/manifest.go

```diff
@@ -182,3 +185,150 @@ func (mr ManifestRouter) DeleteManifest(ctx context.Context, w http.ResponseWrit
+		errMsg := "invalid create application request"
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
+	}
+
+	req := request.ToServiceRequest()
+	createApplicationResponse, err := ar.service.CreateApplication(req)
+	if err != nil {
+		errMsg := "could not create application"
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusInternalServerError)
+	}
+
+	resp := CreateApplicationResponse{Application: createApplicationResponse.Application}
+
+	return framework.Respond(ctx, w, resp, http.StatusCreated)
+}
+
+type GetApplicationResponse struct {
+	ID          string                               `json:"id"`
+	Application applicationsdk.CredentialApplication `json:"application"`
+}
+
+// GetApplication godoc
+// @Summary      Get application
+// @Description  Get application by id
+// @Tags         ApplicationAPI
+// @Accept       json
+// @Produce      json
+// @Param        id   path      string  true  "ID"
+// @Success      200  {object}  GetApplicationResponse
+// @Failure      400  {string}  string  "Bad request"
+// @Router       /v1/manifests/applications/{id} [get]
+func (ar ManifestRouter) GetApplication(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	id := framework.GetParam(ctx, IDParam)
+	if id == nil {
+		errMsg := "cannot get application without ID parameter"
+		logrus.Error(errMsg)
+		return framework.NewRequestErrorMsg(errMsg, http.StatusBadRequest)
+	}
+
+	gotApplication, err := ar.service.GetApplication(manifest.GetApplicationRequest{ID: *id})
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get application with id: %s", *id)
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
+	}
+
+	resp := GetApplicationResponse{
+		ID:          gotApplication.Application.Application.ID,
+		Application: gotApplication.Application,
+	}
+	return framework.Respond(ctx, w, resp, http.StatusOK)
+}
+
+type GetApplicationsResponse struct {
+	Applications []applicationsdk.CredentialApplication `json:"applications"`
+}
+
+// GetApplications godoc
+// @Summary      Get applications
+// @Description  Checks for the presence of a query parameter and calls the associated filtered get method
+// @Tags         ApplicationAPI
+// @Accept       json
+// @Produce      json
+// @Param        issuer   query     string  false  "string issuer"
+// @Param        schema   query     string  false  "string schema"
+// @Param        subject  query     string  false  "string subject"
+// @Success      200      {object}  GetApplicationsResponse
+// @Failure      400      {string}  string  "Bad request"
+// @Failure      500      {string}  string  "Internal server error"
+// @Router       /v1/manifests/applications [get]
+func (ar ManifestRouter) GetApplications(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
```

Comment:
1. querying by status would be useful. we should create a mapping of statuses an application can be in (something like submitted/pending/rejected/fulfilled)
2. the endpoint accepts 3 query parameters as written but I don't see them handled here

Reply:
I left out the pending state stuff for now and updated query parameter comment